### PR TITLE
Add preemption and job restart support for prototyping

### PR DESCRIPTION
**Pull request by shitaoli-db**

### CODE REVIEW

File: ygong/mosaic/submit.py [hunk 1]
```diff
@@ -43,35 +41,9 @@ def _is_local():
...
-            else:
-                print("databricks secret already exists")
-                return
-        from mcli.objects.secrets.create.databricks import DatabricksSecretCreator
-        from mcli.api.secrets.api_create_secret import create_secret
-        s = DatabricksSecretCreator().create(name=databricks_secret_name, host=workspace_url, token=token)
-        print(f"successfully created databricks secret: {databricks_secret_name}")
-        create_secret(s)
-
+        os.environ[config.MCLI_MODE_ENV] = config.MCLIMode.DBX_AWS_STAGING.value
```
Comment (ygong1) [inline ygong/mosaic/submit.py:50]: can you revert this change? It's going to break for my local set up

Comment (shitaoli-db) [reply ygong/mosaic/submit.py:50]: Done.

File: ygong/mosaic/mpt125mConfig.py [hunk 2]
```diff
@@ -1,9 +1,47 @@
 from mcli import RunConfig
 from ygong.mosaic.scaling_config import ScalingConfig
-from typing import Optional
+from typing import Dict, List, Optional
 import os
+import shlex
 # import databricks_genai.api.config as cfg
 
+class WSFSIntegration:
```
Comment (ygong1) [inline ygong/mosaic/mpt125mConfig.py:8]: I did some refactoring and merged your pr. Feel free to develop on the HEAD of the branch
